### PR TITLE
Use a settings instead of a feature flag for LTR.

### PR DESCRIPTION
**Pull request by afoucret**
I figured out that Elasticsearch feature flags are not governed by settings as I was thinking but they are using system properties instead.
It means that enabling the feature flag can be done using `-Des.learning_to_rank_feature_flag_enabled`

This is a pretty big change compared to what I was thinking because it means that the settings can not be set from the config YAML file and it will be impossible to enable the feature on ESS.

This PR is replacing the feature flag with an adhoc setting (`xpack.ml.learning_to_rank.enabled`) that will behave like a soft feature flag.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearningToRankRescorerFeature.java [hunk 1]
```diff
@@ -17,12 +19,22 @@
...
+    /**
+     * This setting behave like a feature flag for the learning to rank feature but can be set by an operator on self-managed environment.
+     * The feature is enabled by default only for snapshots builds.
+     *
+     * See {@link LearningToRankRescorerBuilder}
+     */
+    public static Setting<Boolean> LEARNING_TO_RANK_ENABLED = Setting.boolSetting(
+        "xpack.ml.learning_to_rank.enabled",
+        Build.current().isSnapshot(),
+        Setting.Property.OperatorDynamic,
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearningToRankRescorerFeature.java:31]: If this is what we are going to do, it shouldn't be dynamic.

Comment (afoucret) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearningToRankRescorerFeature.java:31]: Agreed!
I will remove the OperatorDynamic prop.

**Comment by afoucret** [comment]
An alternative PR is available here: https://github.com/elastic/elasticsearch/pull/103333

**Comment by afoucret** [comment]
Closed in favor of https://github.com/elastic/elasticsearch/pull/103333

**Pull request closed without merging**